### ISSUE TITLE
want dask support in CatBoost

### ISSUE THREAD

**Issue by gitclem** [catboost/catboost#1404]
Problem: Dask ( https://docs.dask.org/en/latest/ ) is a very good parallel/distributed data system that replaces pandas.  The people who worked on it, have made it play well with numpy, pandas, scikit-learn, etc.  It would be really nice if CatBoost could take as inputs dask dataframes and dask series objects as well as pandas dataframes and pandas series.  There may be even some clever things that could be done to take advantage of the parallel data.

Unfortunately, now I need to convert my dask data to pandas (very slow!) in order for CB to use the data.
 
catboost version:  0.24.1
Python 3.7.5
Operating System: Ubuntu 18.04.5 LTS (AWS)
CPU: 
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Many thanks! Sounds interesting. Will look into it soon.

**Comment by sephib** [comment]
I'll be happy to contribute to these efforts.   
Here is the [Dask implementation for XGBoost ](https://github.com/dask/dask-xgboost)

**Comment by PyntieHet** [comment]
Could I request Modin.pandas support? Modin is intended as a drop in replacement for pandas built off of Dask so it aligns with above request.

**Comment by qingant** [comment]
Hi guys, any updates on this ?

**Comment by 18kiran12** [comment]
Any updates on this ??

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
dear @18kiran12 ! 
this feature is getting planned right now, and will appear in one of future releases of the library!
we are very happy that you track catboost news ❤️ 
stay tuned!
